src/exec/command.rs

```diff
@@ -7,10 +7,18 @@ use crate::error::print_error;
 use crate::exit_codes::ExitCode;
 
 /// Executes a command.
-pub fn execute_command(mut cmd: Command, out_perm: &Mutex<()>) -> ExitCode {
+pub fn execute_command(mut cmd: Command, out_perm: &Mutex<()>, is_multithread: bool) -> ExitCode {
+    let output;
     // Spawn the supplied command.
-    let output = cmd.output();
-
+    if is_multithread {
+        output = cmd.output();
+    } else {
+        // If running on only one thread, don't buffer output
+        // Allows for viewing and interacting with intermediate command output
+        let child = cmd.spawn().expect("Failed to execute command");
```

Comment:
This looks like it breaks our error handling. What happens for a non-existing command if you run it with `--threads=2` and `--threads=1`? The output should be the same.